### PR TITLE
Add doGet function to request groups from DB and show groups

### PR DESCRIPTION
**Pull request by riku1212**
- Move `DataServlet` from `data` to `servlets` and change name to `GroupServlet`
- Compile utility functions to one place, `JSONUtility`
- Create `doGet` function to read data from database to show groups that the user is in
- Create dropdown to show user's current groups

### CODE REVIEW

File: friendly/src/main/java/com/google/sps/servlets/GroupServlet.java [hunk 1]
```diff
@@ -39,12 +40,35 @@
...
+
+    // user not logged in, do not have groups
+    if (!userService.isUserLoggedIn()){
+        response.sendError(403, "Not authorized to comment.");
+        return;
+    }
+    String ownerId = userService.getCurrentUser().getUserId();
+    
+
+	Query query = new Query("Group")
```
Comment (hhjoel) [inline friendly/src/main/java/com/google/sps/servlets/GroupServlet.java]: I think there's an indentation typo here

File: friendly/src/main/java/com/google/sps/servlets/GroupServlet.java [hunk 2]
```diff
@@ -17,13 +17,14 @@
...
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
-
-import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.PreparedQuery;
```
Comment (hhjoel) [inline friendly/src/main/java/com/google/sps/servlets/GroupServlet.java]: Lexicographic order

File: friendly/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 
-import com.google.gson.Gson;
+import com.google.sps.servlets.JsonUtility;
 import com.google.sps.data.LoginStatus;
```
Comment (hhjoel) [inline friendly/src/main/java/com/google/sps/servlets/LoginServlet.java:21]: Lexicographic order

**Review by hhjoel — APPROVED** [review]
LGTM + 1, other than some trivial code style

File: friendly/src/main/java/com/google/sps/servlets/GroupServlet.java (hunk 1 above)
Comment (riku1212) [reply friendly/src/main/java/com/google/sps/servlets/GroupServlet.java]: ok, done. thanks for pointing it out!

File: friendly/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 3 above)
Comment (riku1212) [reply friendly/src/main/java/com/google/sps/servlets/LoginServlet.java:21]: thanks for pointing this out

File: friendly/src/main/java/com/google/sps/servlets/GroupServlet.java (hunk 2 above)
Comment (riku1212) [reply friendly/src/main/java/com/google/sps/servlets/GroupServlet.java]: thanks for pointing this out

**Pull request merged**